[程序崩溃] [客户端] TUN 入站 + sniff + fakedns

### 完整性要求

- [x] 我读完了 issue 模板中的所有注释，确保填写符合要求。
- [x] 我保证阅读了文档，了解所有我编写的配置文件项的含义，而不是大量堆砌看似有用的选项或默认值。
- [x] 我提供了完整的配置文件和日志，而不是出于自己的判断只给出截取的部分。
- [x] 我搜索了 issues, 没有发现已提出的类似问题。
- [x] 问题在 Release 最新的版本上可以成功复现

### 描述

情景：配置文件中，使用 TUN 入站，并在该入站中配置 sniffing -> enabled: true，并在 dns 模块开启 fakedns
结果：程序崩溃

系统：windows 11 x64
xray版本：
Xray 26.6.1 (Xray, Penetrates Everything.) 94ffd50 (go1.26.3 windows/amd64)

### 重现方式

重现该配置

### 客户端配置

<details><pre><code>
{
  "fakedns": [
    {
      "ipPool": "198.18.0.0/15",
      "poolSize": 65535
    },
    {
      "ipPool": "fc00::/18",
      "poolSize": 65535
    }
  ],
  "inbounds": [
    {
      "tag": "in-tun",
      "protocol": "tun",
      "settings": {
        "name": "tun0",
        "mtu": 1492,
        "gateway": ["10.0.0.1/16", "fdfe::1/64"],
        "dns": ["1.1.1.1"],
        "userLevel": 0,
        "autoSystemRoutingTable": ["0.0.0.0/0", "::/0"],
        "autoOutboundsInterface": "auto"
      },
      "sniffing": {
        "enabled": true
      }
    }
  ]
}
</code></pre></details>

### 服务端配置

NONE

### 客户端日志

Xray 26.6.1 (Xray, Penetrates Everything.) 94ffd50 (go1.26.3 windows/amd64)

2026/06/05 16:04:01.826488 Using existing driver 0.14
2026/06/05 16:04:01.840186 Creating adapter2026/06/05 16:04:02.301508 [Info] proxy/tun: [tun] update interface Wi-Fi 9
2026/06/05 16:04:02.301508 [Info] proxy/tun: tun0 created
2026/06/05 16:04:02.410579 [Info] proxy/tun: tun0 up
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x7ff7533e5340]

goroutine 71 [running]:
net.networkNumberAndMask(0x7ff7530c70cf?)
        net/ip.go:457
net.(*IPNet).Contains(0xd08b4423224?, {0xd08b442e08c, 0x4, 0x1d4633dbf30?})
        net/ip.go:481 +0x25
github.com/xtls/xray-core/app/dns/fakedns.(*Holder).GetDomainFromFakeDNS(...)
        github.com/xtls/xray-core/app/dns/fakedns/fake.go:131
github.com/xtls/xray-core/app/dns/fakedns.(*HolderMulti).GetDomainFromFakeDNS(0xd08b18ce008?, {0x7ff75449b758, 0xd08b4423224})
        github.com/xtls/xray-core/app/dns/fakedns/fake.go:177 +0xaf
github.com/xtls/xray-core/app/dispatcher.newFakeDNSSniffer.func1({0x7ff75449b208, 0xd08b377a150}, {0x1d41daabac0?, 0x6ccf9?, 0x7ff7542ae420?})
        github.com/xtls/xray-core/app/dispatcher/fakednssniffer.go:33 +0xa6
github.com/xtls/xray-core/app/dispatcher.(*Sniffer).SniffMetadata(...)
        github.com/xtls/xray-core/app/dispatcher/sniffer.go:96
github.com/xtls/xray-core/app/dispatcher.sniffer({0x7ff75449b208, 0xd08b377a150}, 0xd08b377a180, 0x0, 0x3)
        github.com/xtls/xray-core/app/dispatcher/default.go:384 +0x2de
github.com/xtls/xray-core/app/dispatcher.(*DefaultDispatcher).DispatchLink(0xd08b1ca4a50, {0x7ff75449b208, 0xd08b377a150}, {{0x7ff75449b758?, 0xd08b4423224?}, 0x2e80?, 0xd08?}, 0xd08b1b34960)
        github.com/xtls/xray-core/app/dispatcher/default.go:350 +0x605
github.com/xtls/xray-core/proxy/tun.(*Handler).HandleConnection(0xd08b1a9aa50, {0x7ff7544ab8b0, 0xd08b1a4b940}, {{0x7ff75449b758, 0xd08b4423224}, 0x89, 0x3})
        github.com/xtls/xray-core/proxy/tun/handler.go:163 +0xb1d
created by github.com/xtls/xray-core/proxy/tun.(*udpConnectionHandler).HandlePacket in goroutine 67
        github.com/xtls/xray-core/proxy/tun/udp_fullcone.go:67 +0x445

### 服务端日志

NONE

--- a/app/proxyman/inbound/always.go
+++ b/app/proxyman/inbound/always.go
@@ -170,6 +170,12 @@ func NewAlwaysOnInboundHandler(ctx context.Context, tag string, receiverConfig *
 
 // Start implements common.Runnable.
 func (h *AlwaysOnInboundHandler) Start() error {
+	// for inbound without worker (TUN)
+	if run, ok := h.proxy.(common.Runnable); ok {
+		if err := run.Start(); err != nil {
+			return errors.New("failed to start proxy").Base(err)
+		}
+	}
 	for _, worker := range h.workers {
 		if err := worker.Start(); err != nil {
 			return err

--- a/proxy/tun/handler.go
+++ b/proxy/tun/handler.go
@@ -40,10 +40,11 @@ type ConnectionHandler interface {
 // Handler implements ConnectionHandler
 var _ ConnectionHandler = (*Handler)(nil)
 
+// Handler implements common.Runnable
+var _ common.Runnable = (*Handler)(nil)
+
 // Init the Handler instance with necessary parameters
 func (t *Handler) Init(ctx context.Context, pm policy.Manager, dispatcher routing.Dispatcher) error {
-	var err error
-
 	// Retrieve tag and sniffing config from context (set by AlwaysOnInboundHandler)
 	if inbound := session.InboundFromContext(ctx); inbound != nil {
 		t.tag = inbound.Tag
@@ -56,6 +57,10 @@ func (t *Handler) Init(ctx context.Context, pm policy.Manager, dispatcher routin
 	t.policyManager = pm
 	t.dispatcher = dispatcher
 
+	return nil
+}
+
+func (t *Handler) Start() error {
 	tunName := t.config.Name
 	tunInterface, err := NewTun(t.config)
 	if err != nil {
@@ -92,7 +97,7 @@ func (t *Handler) Init(ctx context.Context, pm policy.Manager, dispatcher routin
 
 	tunStackOptions := StackOptions{
 		Tun:         tunInterface,
-		IdleTimeout: pm.ForLevel(t.config.UserLevel).Timeouts.ConnectionIdle,
+		IdleTimeout: t.policyManager.ForLevel(t.config.UserLevel).Timeouts.ConnectionIdle,
 	}
 	tunStack, err := NewStack(t.ctx, tunStackOptions, t)
 	if err != nil {
@@ -167,7 +172,7 @@ func (t *Handler) HandleConnection(conn net.Conn, destination net.Destination) {
 
 // Close implements common.Closable.
 func (t *Handler) Close() error {
-	return errors.Combine(t.stack.Close(), t.tun.Close())
+	return errors.Combine(common.CloseIfExists(t.stack), common.CloseIfExists(t.tun))
 }
 
 // Network implements proxy.Inbound

